Browser extension section improvements (#14513)
sourcegraph/sourcegraph: doc/integration/browser_extension.md

To install this search engine shortcut manually, and for more information, see "[Browser search engine shortcuts](browser_search_engine.md)".

## Make it work on your code host

- [github.com](https://github.com/) - no action required.

> - GitHub Enterprise, GitLab, Bitbucket Server and Phabricator - additional permissions are needed.

Reviewer: I think the term "additional" is not accurate because we don't need an additional kind of permission compared to what we already have on github.com, we just need permissions to do anything at all on those pages (without permission we can't do anything).
Here's a stab at a better wording:
```suggestion
- GitHub Enterprise, GitLab, Bitbucket Server and Phabricator - you need to grant the extension permissions first.
```

1. Click "Enable Sourcegraph on this domain".
1. Click "Allow" in the permissions request popup.

## Make it work for private code

![Configure Sourcegraph Instance](https://sourcegraphstatic.com/ConfigureSourcegraphInstanceUse.gif)

> By default, the browser extension communicates with [Sourcegraph.com](https://sourcegraph.com), which has only public code.

Reviewer: ```suggestion
By default, the browser extension communicates with [sourcegraph.com](https://sourcegraph.com), which has only public code.
```

To use the browser extension with your private repositories, you need to set up a private Sourcegraph instance and connect it to the extension.

> Follow the instructions:

Reviewer: ```suggestion
Follow these instructions:
```

1. Click "Allow" in the permissions request popup.

## Make it work for private code

<video class="theme-dark-only" width="1764" height="1390" autoplay loop muted playsinline style="width: 100%; max-width: 50rem">
  <source src="https://storage.cloud.google.com/sourcegraph-assets/code-host-integration/PrivateInstanceDark.webm" type="video/webm">
  <source src="https://storage.cloud.google.com/sourcegraph-assets/code-host-integration/PrivateInstanceDark.mp4" type="video/mp4">
  <p>Configure browser extension for your private Sourcegraph instance</p>
</video>
<video class="theme-light-only" width="1764" height="1390" autoplay loop muted playsinline style="width: 100%; max-width: 50rem">
  <source src="https://storage.cloud.google.com/sourcegraph-assets/code-host-integration/PrivateInstanceLight.webm" type="video/webm">
  <source src="https://storage.cloud.google.com/sourcegraph-assets/code-host-integration/PrivateInstanceLight.mp4" type="video/mp4">
  <p>Configure browser extension for your private Sourcegraph instance</p>
</video>

By default, the browser extension communicates with [sourcegraph.com](https://sourcegraph.com), which has only public code.

> To use the browser extension with your private repositories, you need to set up a private Sourcegraph instance and connect it to the extension.

Reviewer: ```suggestion
To use the browser extension with your private repositories, you need to set up a private Sourcegraph instance and connect the extension to it.
```
